corehq/apps/zapier/tests.py

```diff
@@ -396,3 +397,122 @@ def test_case_fields(self):
+        cls.domain = cls.domain_object.name
+        cls.user = WebUser.create(cls.domain, 'test', '******')
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete()
+        cls.domain_object.delete()
+        super(TestZapierCreateCaseAction, cls).tearDownClass()
+
+    def test_create_case(self):
+        factory = RequestFactory()
+        request = factory.post(self.test_url,
+                              data=json.dumps(self.data),
+                              content_type='application/json')
+
+        status = ZapierCreateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+
+        case_id = self.accessor.get_case_ids_in_domain()
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('test1', case.get_case_property('name'))
+        self.assertEqual('11', case.get_case_property('price'))
+        self.assertEqual('watermelon', case.get_case_property('type'))
+        self.assertEqual('test_user', case.get_case_property('owner_id'))
+        self.accessor.soft_delete_cases(case_id)
+
+    def test_update_case(self):
+        factory = RequestFactory()
+        request = factory.post(self.test_url,
+                              data=json.dumps(self.data),
+                              content_type='application/json')
+        request.user = self.user
+
+        status = ZapierCreateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+        case_id = self.accessor.get_case_ids_in_domain()
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('11', case.get_case_property('price'))
+
+        data = {'case_name': 'test1', 'price': '15', 'case_id': case_id[0]}
+        request = factory.post(self.test_url,
+                               data=json.dumps(data),
+                               content_type='application/json')
+
+        status = ZapierUpdateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('15', case.get_case_property('price'))
+
+        self.accessor.soft_delete_cases(case_id)
+
+    def test_update_case_does_not_exist(self):
+        factory = RequestFactory()
+        data = {'case_name': 'test1', 'price': '15', 'case_id': 'fake_id'}
+        request = factory.post(self.test_url,
+                               data=json.dumps(data),
+                               content_type='application/json')
+
+        status = ZapierUpdateCase().post(request)
+        self.assertEqual(status.status_code, 403)
+
+    def test_update_case_wrong_domain(self):
+        factory = RequestFactory()
+        request = factory.post(self.test_url,
+                               data=json.dumps(self.data),
+                               content_type='application/json')
+        ZapierCreateCase().post(request)
+
+        factory = RequestFactory()
+        data = {'case_name': 'test1', 'price': '15', 'case_id': 'fake_id'}
```

Comment:
I think this test is also demonstrating that updating a case that does not exist. On this line you might have meant to use the id of the case that was just created?